src/test/codegen/link_section.rs

```diff
@@ -12,17 +12,11 @@
 
 #![crate_type = "lib"]
 
-// CHECK: @VAR1 = constant <{ [4 x i8] }> <{ [4 x i8] c"\01\00\00\00" }>, section ".test_one"
+// CHECK: @VAR1 = constant i32 1, section ".test_one"
```

Comment:
@eddyb The types of static are changed for some values. I assume all uses of statics use bitcasts anyway?

Reply:
This seems wrong, why would a `static` go through anything other than an `Allocation`?

Reply:
Here is the code which converts constants into LLVM values: https://github.com/rust-lang/rust/pull/50249/files#diff-823b4422fff3f3018e53dfb6dc0a1fb3R84

It does not seem useful to convert these back into an `Allocation`.

Reply:
~~`static`s should *never* be anything other than `Allocation`, so this seems wrong. Evaluating the initializer of a `static` should not be done the same way as evaluating a `const`. Only the former should produce a value, whereas a `static` should *already* have an allocation, and the evaluation should "fill it in"~~

Hang on, I got a bit turned around. The value in this case is the initializer, so `ByRef` indicates *not* the "allocation of the static" but the "contents" - it just happens to use `&Allocation`, which has become confusing.

In which case, if no information is obtained from that `Allocation` *other* than the bytes and the relocations, it seems fine (I just checked and this is indeed the case).

Can you inline the body of `global_initializer` into `const_value_to_llvm` and rename the latter to `global_initializer`? At least that way it can't be easily confused with the case where `ByRef` actually requires indirection.

Reply:
`global_initializer` is still used elsewhere. I've renamed it to `const_alloc_to_llvm`.